File: journal-monitor/internal/model/journal.go
package model

import "time"

type Journal struct {
	ID            string    `json:"id"`
	Name          string    `json:"name"`
	Slug          string    `json:"slug"`
	SourceType    string    `json:"source_type"` // "rss" | "arxiv" | "crossref"
	SourceURL     string    `json:"source_url"`
	FetchInterval time.Duration `json:"fetch_interval"`
	IsActive      bool      `json:"is_active"`
	CreatedBy     *string   `json:"created_by,omitempty"`
	CreatedAt     time.Time `json:"created_at"`
}

type JournalRequest struct {
	ID          string     `json:"id"`
	UserID      string     `json:"user_id"`
	JournalName string     `json:"journal_name"`
	SourceURL   string     `json:"source_url"`
	Status      string     `json:"status"` // "pending" | "approved" | "rejected"
	CreatedAt   time.Time  `json:"created_at"`
	ReviewedAt  *time.Time `json:"reviewed_at,omitempty"`
}

type CreateJournalRequest struct {
	JournalName string `json:"journal_name" binding:"required"`
	SourceURL   string `json:"source_url" binding:"required,url"`
}

File: app/models/journal.py
import datetime as dt
import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, ForeignKey, Interval, String, Text, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.db import Base


class Journal(Base):
    __tablename__ = "journals"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, server_default=func.gen_random_uuid()
    )
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    slug: Mapped[str] = mapped_column(String(255), unique=True, nullable=False)
    source_type: Mapped[str] = mapped_column(String(20), nullable=False)
    source_url: Mapped[str] = mapped_column(Text, nullable=False)
    description: Mapped[str] = mapped_column(Text, nullable=False, server_default="")
    fetch_interval: Mapped[dt.timedelta] = mapped_column(
        Interval, nullable=False, server_default="30 minutes"
    )
    is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, server_default="true")
    created_by: Mapped[uuid.UUID | None] = mapped_column(
        UUID(as_uuid=True), ForeignKey("users.id"), nullable=True
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, server_default=func.now()
    )


class JournalRequest(Base):
    __tablename__ = "journal_requests"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, server_default=func.gen_random_uuid()
    )
    user_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False
    )
    journal_name: Mapped[str] = mapped_column(String(255), nullable=False)
    source_url: Mapped[str] = mapped_column(Text, nullable=False)
    status: Mapped[str] = mapped_column(String(20), nullable=False, server_default="pending")
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, server_default=func.now()
    )
    reviewed_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
